src/FieldManager.C

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+#include "FieldManager.h"
+#include "stk_mesh/base/MetaData.hpp"
+
+namespace sierra {
+namespace nalu {
+
+FieldManager::FieldManager(stk::mesh::MetaData& meta, int numStates)
+  : meta_(meta), numStates_(numStates)
+{
+}
+
+bool
+FieldManager::field_exists(std::string name)
+{
+  auto definition = FieldRegistry::query(numStates_, name);
+
+  return std::visit(
+    [&](auto def) -> bool {
+      return meta_.get_field<typename decltype(def)::FieldType>(
+               def.rank, name) != nullptr;
+    },
+    definition);
```

Comment:
I'm not sure if that will work.  The `FieldDefTypes` is a `std::variant` type.  This is done so we can store all the field types in a single map and have one map in the registry. Otherwise we would need multiple maps for each type.  So we actually don't know the type inside the variant that will match our query.   `std::visit` loops over all the types in the variant and we store the type data we need based on the entry in the variant.  So the other option I'm aware of is to use `std::get` for each type and return the one that is not `null`, but that is just essentially what `std::visit` is doing.

Reply:
oh ok, yeah I didn't understand what exactly `std::visit` was doing.  This makes sense now.